cognite/client/_api/datapoints.py

```diff
@@ -99,7 +101,7 @@ def retrieve(
             limit=limit,
         )
         dps_list = fetcher.fetch(query)
-        if is_single_id:
+        if is_single_id and not ignore_unknown_ids:
             return dps_list[0]
```

Comment:
```
if is_single_id:
    if ignore_unknown_ids is True:
        return None
    return dps_list[0]
```
And change the return type hint to `Union[Datapoints, DatapointsList, None]`

Reply:
having two return types for retrieve is already a pain, do you really think three different ones will help?

Reply:
to follow up:
2 requested, ignoreunknown False -> DatapointsList of length 2
single requested, ignoreunknown False -> Datapoints
2 requested, 1 found, ignoreunknown True -> DatapointsList of length 1
2 requested, 0 found, ignoreunknown True -> DatapointsList of length 0
single requested, 0 found ignoreunknown True -> new behaviour: None ?
agreed?

Reply:
The nested if there should of course be `if ignore_unknown_ids and len(dps_list) == 0`. Sorry..

IMO it is better to have behaviour consistent with other methods for single retrieval than returning an empty list upon single retrieval not found.

Reply:
I fixed it in the last commit. It's a bit painful in retrieve_dataframe to handle 3 return types, but I think it's ok now